app/controllers/api/v1/sources_controller.rb

```diff
@@ -15,6 +15,23 @@ def create
 
         render :json => source, :status => :created, :location => instance_link(source)
       end
+
+      def check_availability
+        source = Source.find(params[:source_id])
+
+        Sources::Api::Messaging.client.publish_topic(
+          :service => "platform.topological-inventory.operations-#{source.source_type.name}",
+          :event   => "Source.availability_check",
+          :payload => {
+            :params => {
+              :source_id       => source.id.to_s,
+              :external_tenant => source.tenant.external_tenant
+            }
```

Comment:
We should create a task so the requester can track the progress of the availability check, pass the task ID in as a param, and return the ID to the caller

Reply:
Scratch that, we're going to hold off on the UI blocking on the availability check